src/ompl/geometric/planners/informedtrees/bitstar/src/ImplicitGraph.cpp

```diff
@@ -527,6 +527,9 @@ namespace ompl
                         // Add it back to the vector
                         startVertices_.push_back(*prunedStartIter);
 
+                        // Add as a sample
+                        this->addToSamples(*prunedStartIter);
```

Comment:
This part of the code is "reactivating" previously removed starts that, thanks to the addition of a new goal, now could be useful.

But (if I remember correctly) everything in `startVertices_` is considered part of the search tree by definition (even if it's just a singular root), so I don't think they should also be added to the samples...

Reply:
I'm pretty sure not, Line 537 right below has:
```
// Add the vertex to the set of vertices.
this->registerAsVertex(*prunedStartIter);
```
and I'm pretty sure the Vertices and Samples are supposed to be mutually exclusive sets... I think.

Reply:
Oh, [Lines 430-431](https://github.com/robotic-esp/ompl/blob/eeebb7e3a29ee50057e41cea573815c1f6121bf3/src/ompl/geometric/planners/informedtrees/bitstar/src/ImplicitGraph.cpp#L430-L431) add a new start to both samples and vertices,
```
this->addToSamples(startVertices_.back());
this->registerAsVertex(startVertices_.back());
```
so that would suggest this is right...? I don't remember this code very well anymore.

Reply:
Looking at what when a sample is first connected as a new vertex in [`BITstar::addEdge`](https://github.com/robotic-esp/ompl/blob/eeebb7e3a29ee50057e41cea573815c1f6121bf3/src/ompl/geometric/planners/informedtrees/src/BITstar.cpp#L808C16-L848) and what [`BITstar::ImplicitGraph::registerAsVertex`](https://github.com/robotic-esp/ompl/blob/eeebb7e3a29ee50057e41cea573815c1f6121bf3/src/ompl/geometric/planners/informedtrees/bitstar/src/ImplicitGraph.cpp#L736C14-L756) actually does, it looks to me like all vertices should be, and stay, in the set of samples.

So this is in fact right, sorry for the confusion!